### ISSUE TITLE
can't compile on ubuntu-18.04

### ISSUE THREAD

**Issue by storypku** [google/orbit#1103]
Hello team,
    When I checked out the latest `master` branch of  `orbit` on ubuntu 18.04

```
Prompt> uname -a
Linux myhostname 5.4.0-42-generic #46~18.04.1-Ubuntu SMP Fri Jul 10 07:21:24 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

and tries to compile it by running `bash bootstrap-orbit.sh` with gcc-9 as the default compiler, the following errors occured.

```
[ 94%] Building CXX object OrbitQt/CMakeFiles/OrbitQt.dir/OrbitQt_autogen/KGPS6WV3FN/qrc_orbiticons.cpp.o
[ 94%] Building CXX object OrbitQt/CMakeFiles/OrbitQt.dir/topdownwidget.cpp.o
In file included from /usr/include/x86_64-linux-gnu/qt5/QtGui/QOpenGLFunctions:1,
                 from /apollo/orbit/OrbitQt/orbitglwidget.h:8,
                 from /apollo/orbit/OrbitQt/orbittreeview.h:13,
                 from /apollo/orbit/OrbitQt/orbittreeview.cpp:11:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglfunctions.h:49:2: warning: #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined [-Wcpp]
 #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined
  ^~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglfunctions.h:50:2: warning: #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h [-Wcpp]
 #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h
  ^~~~~~~
In file included from /usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglfunctions.h:55,
                 from /usr/include/x86_64-linux-gnu/qt5/QtGui/QOpenGLFunctions:1,
                 from /apollo/orbit/OrbitQt/orbitglwidget.h:8,
                 from /apollo/orbit/OrbitQt/orbittreeview.h:13,
                 from /apollo/orbit/OrbitQt/orbittreeview.cpp:11:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglcontext.h:55:2: warning: #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined [-Wcpp]
 #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined
  ^~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglcontext.h:56:2: warning: #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h [-Wcpp]
 #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h
  ^~~~~~~
/apollo/orbit/OrbitQt/MainThreadExecutorImpl.cpp: In member function 'virtual void {anonymous}::MainThreadExecutorImpl::Schedule(std::unique_ptr<Action>)':
/apollo/orbit/OrbitQt/MainThreadExecutorImpl.cpp:23:82: error: no matching function for call to 'QMetaObject::invokeMethod(QCoreApplication*, {anonymous}::MainThreadExecutorImpl::Schedule(std::unique_ptr<Action>)::<lambda()>)'
                             [action = std::move(action)]() { action->Execute(); });
                                                                                  ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qobject.h:46,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qcoreapplication.h:46,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QCoreApplication:1,
                 from /apollo/orbit/OrbitQt/MainThreadExecutorImpl.cpp:7:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobjectdefs.h:406:17: note: candidate: 'static bool QMetaObject::invokeMethod(QObject*, const char*, Qt::ConnectionType, QGenericReturnArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument)'
     static bool invokeMethod(QObject *obj, const char *member,
                 ^~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobjectdefs.h:406:17: note:   candidate expects 14 arguments, 2 provided
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobjectdefs.h:420:24: note: candidate: 'static bool QMetaObject::invokeMethod(QObject*, const char*, QGenericReturnArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument)'
     static inline bool invokeMethod(QObject *obj, const char *member,
                        ^~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobjectdefs.h:420:24: note:   candidate expects 13 arguments, 2 provided
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobjectdefs.h:437:24: note: candidate: 'static bool QMetaObject::invokeMethod(QObject*, const char*, Qt::ConnectionType, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument)'
     static inline bool invokeMethod(QObject *obj, const char *member,
                        ^~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobjectdefs.h:437:24: note:   candidate expects 13 arguments, 2 provided
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobjectdefs.h:454:24: note: candidate: 'static bool QMetaObject::invokeMethod(QObject*, const char*, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument)'
     static inline bool invokeMethod(QObject *obj, const char *member,
                        ^~~~~~~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtCore/qobjectdefs.h:454:24: note:   no known conversion for argument 2 from '{anonymous}::MainThreadExecutorImpl::Schedule(std::unique_ptr<Action>)::<lambda()>' to 'const char*'
OrbitQt/CMakeFiles/OrbitQt.dir/build.make:134: recipe for target 'OrbitQt/CMakeFiles/OrbitQt.dir/MainThreadExecutorImpl.cpp.o' failed
make[2]: *** [OrbitQt/CMakeFiles/OrbitQt.dir/MainThreadExecutorImpl.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /usr/include/x86_64-linux-gnu/qt5/QtGui/QOpenGLFunctions:1,
                 from /apollo/orbit/OrbitQt/orbitglwidget.h:8,
                 from /apollo/orbit/OrbitQt/orbittreeview.h:13,
                 from /apollo/orbit/OrbitQt/orbitdataviewpanel.h:12,
                 from /apollo/orbit/build_default_relwithdebinfo/OrbitQt/OrbitQt_autogen/include/ui_processlauncherwidget.h:19,
                 from /apollo/orbit/OrbitQt/processlauncherwidget.cpp:13:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglfunctions.h:49:2: warning: #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined [-Wcpp]
 #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined
  ^~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglfunctions.h:50:2: warning: #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h [-Wcpp]
 #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h
  ^~~~~~~
In file included from /usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglfunctions.h:55,
                 from /usr/include/x86_64-linux-gnu/qt5/QtGui/QOpenGLFunctions:1,
                 from /apollo/orbit/OrbitQt/orbitglwidget.h:8,
                 from /apollo/orbit/OrbitQt/orbittreeview.h:13,
                 from /apollo/orbit/OrbitQt/orbitdataviewpanel.h:12,
                 from /apollo/orbit/build_default_relwithdebinfo/OrbitQt/OrbitQt_autogen/include/ui_processlauncherwidget.h:19,
                 from /apollo/orbit/OrbitQt/processlauncherwidget.cpp:13:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglcontext.h:55:2: warning: #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined [-Wcpp]
 #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined
  ^~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglcontext.h:56:2: warning: #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h [-Wcpp]
 #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h
  ^~~~~~~
In file included from /usr/include/x86_64-linux-gnu/qt5/QtGui/QOpenGLFunctions:1,
                 from /apollo/orbit/OrbitQt/orbitglwidget.h:8,
                 from /apollo/orbit/OrbitQt/orbittreeview.h:13,
                 from /apollo/orbit/OrbitQt/orbitdataviewpanel.h:12,
                 from /apollo/orbit/OrbitQt/orbitsamplingreport.h:12,
                 from /apollo/orbit/OrbitQt/orbitmainwindow.cpp:37:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglfunctions.h:49:2: warning: #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined [-Wcpp]
 #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined
  ^~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglfunctions.h:50:2: warning: #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h [-Wcpp]
 #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h
  ^~~~~~~
In file included from /usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglfunctions.h:55,
                 from /usr/include/x86_64-linux-gnu/qt5/QtGui/QOpenGLFunctions:1,
                 from /apollo/orbit/OrbitQt/orbitglwidget.h:8,
                 from /apollo/orbit/OrbitQt/orbittreeview.h:13,
                 from /apollo/orbit/OrbitQt/orbitdataviewpanel.h:12,
                 from /apollo/orbit/OrbitQt/orbitsamplingreport.h:12,
                 from /apollo/orbit/OrbitQt/orbitmainwindow.cpp:37:
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglcontext.h:55:2: warning: #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined [-Wcpp]
 #warning qopenglfunctions.h is not compatible with GLEW, GLEW defines will be undefined
  ^~~~~~~
/usr/include/x86_64-linux-gnu/qt5/QtGui/qopenglcontext.h:56:2: warning: #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h [-Wcpp]
 #warning To use GLEW with Qt, do not include <qopengl.h> or <QOpenGLFunctions> after glew.h
  ^~~~~~~
CMakeFiles/Makefile2:1865: recipe for target 'OrbitQt/CMakeFiles/OrbitQt.dir/all' failed
make[1]: *** [OrbitQt/CMakeFiles/OrbitQt.dir/all] Error 2
Makefile:94: recipe for target 'all' failed
make: *** [all] Error 2
ERROR: conanfile.py (OrbitProfiler/None): Error in build() method, line 143
	cmake.build()
	ConanException: Error 2 while executing cmake --build '/apollo/orbit/build_default_relwithdebinfo' '--' '-j20'
```

My steps are listed as follows:

1) Install conan  via `sudo -H pip3 install conan` ( I have symlinked `/usr/bin/python` to point to `python3` )
2) Install gcc-9 and make it the default gcc/g++ toolchain by running the following commands

```
sudo add-apt-repository ppa:ubuntu-toolchain-r/test
sudo apt-get -y update \
  && sudo apt-get -y install \
    gcc-7 \
    g++-7 \
    gcc-9 \
    g++-9 \
  && sudo apt-get clean

sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 90
sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-9 90

sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 70
sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-7 70
```

3) `./bootstrap-orbit.sh` # inside orbit git repo, it failed, so I added step 4)
4)  `conan profile update settings.compiler.libcxx=libstdc++11 default`
5) Run `./bootstrap-orbit.sh` again.

After step 5), It shows the error messages listed above.

The same problem with the latested tag, namely https://github.com/google/orbit/archive/1.51.tar.gz.

Can anyone help with that? Thanks.

**Comment by beckerhe** [comment]
Hi Jiaming,
what you're doing seems correct. The problem is likely with a Qt version that is too old.
I've done some research and it seems you need at least version 5.12.4.

Any chances that you can update your version of Ubuntu? Both 19.10 (eoan) and 20.04LTS (focal) should be fine.

Cheers,
Henning

**Comment by storypku** [comment]
> Hi Jiaming,
> what you're doing seems correct. The problem is likely with a Qt version that is too old.
> I've done some research and it seems you need at least version 5.12.4.
> 
> Any chances that you can update your version of Ubuntu? Both 19.10 (eoan) and 20.04LTS (focal) should be fine.
> 
> Cheers,
> Henning

Thank you for your reply and point to me the possible cause/fix of this problem.
I had previously tested with Qt version 5.12.2 from official Qt,  the same error exists.  Maybe I should try with a more recent Qt version.

For my use case, I have to use ubuntu 18.04.02 for my test, and I think it's sufficient as the kernel version is already 5.4.0+.

Regards,
          Jiaming Liu

**Comment by beckerhe** [comment]
> > Hi Jiaming,
> > what you're doing seems correct. The problem is likely with a Qt version that is too old.
> > I've done some research and it seems you need at least version 5.12.4.
> > Any chances that you can update your version of Ubuntu? Both 19.10 (eoan) and 20.04LTS (focal) should be fine.
> > Cheers,
> > Henning
> 
> Thank you for your reply and point to me the possible cause/fix of this problem.
> I had previously tested with Qt version 5.12.2 from official Qt, the same error exists. Maybe I should try with a more recent Qt version.
> 
> For my use case, I have to use ubuntu 18.04.02 for my test, and I think it's sufficient as the kernel version is already 5.4.0+.
> 
> Regards,
> Jiaming Liu

The kernel should be fine. 5.12.2 is not enough though, see here: https://bugreports.qt.io/browse/QTBUG-69683 That's the bug that is giving you trouble. It was fixed in 5.12.4.
